src/libs/ajax/leonardo/models/runtime-models.ts

```diff
@@ -114,7 +132,47 @@ export interface GetRuntimeItem {
   patchInProgress: boolean;
 }
 
-export type Runtime = GetRuntimeItem | ListRuntimeItem;
+export interface GetRuntimeItem {
+  id: number;
+  runtimeName: string;
+  googleProject: string;
+  cloudContext: CloudContext;
+  serviceAccount: string;
+  asyncRuntimeFields: AsyncRuntimeFields | null;
+  auditInfo: AuditInfo;
+  runtimeConfig: RuntimeConfig;
+  proxyUrl: string;
+  status: LeoRuntimeStatus;
+  labels: RuntimeLabels;
+  userScriptUri: string | null;
+  startUserScriptUri: string | null;
+  jupyterUserScriptUri: string | null;
+  jupyterStartUserScriptUri: string | null;
+  errors: RuntimeError[];
+  userJupyterExtensionConfig: UserJupyterExtensionConfig | null;
+  autopauseThreshold: number;
+  defaultClientId: string | null;
+  runtimeImages: LeoRuntimeImage[];
+  scopes: string[];
+  customEnvironmentVariables: Record<string, any>;
+  diskConfig: DiskConfig | null;
+  patchInProgress: boolean;
+}
+
+const optionalFields = new Set([
```

Comment:
Hmm... I don't think this is necessary again. Should be safe to do `_.omitBy(obj, v => v === null);`
@msilva-broad, thoughts about sanitizing nulls out of this object?

To re-cap, the backend swagger docs specify these fields as optional. Unfortunately, the encoding of this object into JSON includes optional fields as `null` in the payload. Here, we bridge the gap between the backend's definition of optional to be in line with typescripts definition of optional by dropping `null` values from the object.

Reply:
ignore once again